client_option.go

```diff
@@ -0,0 +1,63 @@
+package gax
+
+type ClientOption interface {
+	Resolve(*clientSettings)
+}
+
+type clientOptions []ClientOption
+
+func (opts clientOptions) Resolve(s *clientSettings) *clientSettings {
+	for _, opt := range opts {
+		opt.Resolve(s)
+	}
+	return s
+}
+
+type clientSettings struct {
+	apiName    string
+	apiVersion string
+	endpoint   string
+	scopes     []string
+}
+
+type withAPIName string
+
+func (w withAPIName) Resolve(s *clientSettings) {
+	s.apiName = string(w)
+}
+
+func WithAPIName(apiName string) ClientOption {
```

Comment:
It could, but what's done here is fine too, and probably renders a little better in the docs. It also more easily permits more stateful options.